### PR TITLE
Feature/ietf007 iab site updates fab

### PR DESCRIPTION
**Pull request by CliffRobinson**
Merging FED changes to the IAB home page into @AndrewCalderSpringload 's BED branch.

Video of page behaviours:

https://user-images.githubusercontent.com/37293617/220480551-e7e69c52-d657-4573-9ee3-f40f2bb26e6f.mov

### CODE REVIEW

File: ietf/home/templates/includes/post_iab.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            {% image post.feed_image height-245 as post_image %}
+            <img class="blog-post-image-iab u-obj-cover w-100 h-100" src="{{ post_image.url }}" alt="{{ post.main_image.title }}">
+        {% endif %}
+    </div>
+    {# col-* classes include positioning, position static cancels that and allows the stretched link to cover the image #}
+    <div class="card-body card-iab">
+        <h3 class="mb-3 h5"><a class="stretched-link" href="{{ post.url }}">{{ post.title }}</a></h3>
+        <p class="card-text-iab">{{ post.introduction }}</p>
+        <div class="d-flex justify-content-between align-items-center date-text-iab">
+            <small>
```
Comment (liamjohnston) [inline ietf/home/templates/includes/post_iab.html:15]: I haven't seen the `<small>` tag used in years, though I can't see anywhere saying not to use it. It's semantically fine. Though if you wanted more fine-grained control over what size this actually renders at (to match the design) I'd suggest just adding BEM classes to the `<p>`s.

File: ietf/static/img/iab-logo.svg [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
...oKKW62whLZLe0tGQnJibevHr16g82b4aD770Hr6bT6VcrKyvr9u3bl2shrRIiou4bN27kFmSxHnVGUcnFiq/uLl25Rn1HlgSIyEsCm28KaNfGh7Y8AtKn364jKv0cWkTMrVUPLWK2h+tiQrevE9geiFRGlDiEREU0Pj5+S1F4VokgnU5fUi0kkYjImGIRUkIiIiojfl0Ozf/0kRK0+Msr6qdG+4qJCoA1kZKsbTYVDamMRBULnbZvJ0qNDZFM00Rcu3btSDqd7m5ublYyZXMSEVjVLDOZzN3T09Njzz33HMzMzEQ+JicRWWMaMRfnt6qcsiU0R9TL1b+uJiIQJZtF+Rm+mBj+e9YqX4vJ5oMe62A7LboUbV/njHyiTEagUEhuIkJUCslNRGRMyoSUR5d48JUDPTcGJO+vspl29UpYSCmqT21Xs1pjg1IZgQIheRURokJIXkVExhS5kPLwWjORgDD6aRMc/DSBPExKrYI11dtL+tb7lRtPk2D7UXcdKTiUywgiFJJfESFRCsmviMiYIhNSHoioieRxcA0PymWK5F9EbXzAeg9tCz3sIJz9IcTBbx+FtF+fTfOPsgS2CJlJ7aAioshOagcVETcmqUntBIvI6+Uge8m0ij9rJQKTxw3Wa5FRroA8foZdAlvUqqeNW3ZAmdJn0/wRS2SEyIqQZIgIJEdIMkQEkiOkPC4Dgu2T+UtDphwOdL6Nzii3ABJF1OujOy3POe4zkb4Qn7lsiTUyQsJESLJERAkbIckSETemUBHSMqhH5KdVDkY7Mkt74GfmY7mTRJAIGUFAIUUhIiSokKIQERlTICHpwmiafCDWaRrF75QtShFBwClblCKCgFM2LSJNvpAYGYEPIUUtIsSPkKIWERmTZyFpEWnyiUTJCDwISZWIEC9CUiUiMiZXIWkRafKNxMkIHISkWkSIk5BUi4iMyVZIWkSafCQxCWwRmNQ+fPhwqr6+Hrq7u5WLiEKT2g899BBcv34dXn/9deUi4saUS2rfVveN0rKazTDz8S+1iDR5SaJlBEsHWzsA9ABAPQD0m6bZH/N4Kq3xsNqRI9aYBmMeE/tuDgAA+65OsfumaV6Jc0wajV8SLyONRrM8SGTOSKPRLD+0jDQaTSLQMtJoNIlAy0ij0SQCLSONRpMItIw0Gk38AMD/A//VoOoJFDCLAAAAAElFTkSuQmCC"/>
```
Comment (liamjohnston) [inline ietf/static/img/iab-logo.svg]: Hmm.

https://springload.slack.com/archives/C3JHNA6T0/p1668579172998559

This should be either a proper SVG or just use a png.

File: ietf/static/img/linkedin-icon.svg [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="rgb(248, 249, 250)" class="bi bi-linkedin" viewBox="0 0 16 16">
```
Comment (liamjohnston) [inline ietf/static/img/linkedin-icon.svg:1]: For single-colour icons it's usually preferable to set the fill(s) as `currentColor`, then use CSS to set/change the colour. That way you can change it easily on hover or other contexts.

File: ietf/static_src/css/iab.scss [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+$blue: #0b8cc5; // (11, 140, 97)
+$blue-2: #004f73; // (0, 79, 115)
+$navy: #002d3c; // (0, 45, 60)
+$yellow: #ffd13e; // (255, 209, 62)
+
+$grey-1: #f7fafc; // (247, 250, 252)
+$grey-2: #f0f6fa; // (240, 246, 250)
+$grey-3: #e6edf2; // (230, 237, 242)
+$grey-4: #d9e0e5; // (217, 224, 229)
+$grey-5: #cbd2d6; // (230, 210, 214)
```
Comment (liamjohnston) [inline ietf/static_src/css/iab.scss]: We usually prefer CSS variables nowadays. Or do you have a IE support requirement?

File: ietf/static_src/css/iab.scss [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    // copied
+    background-position: 50%;
+    background-repeat: no-repeat;
+    background-size: cover;
+    position: relative;
+    &:before {
+        // second color is $brand-blue with 0% opacity
+        background-image: linear-gradient(
+            180deg,
+            $brand-blue 17.96%,
```
Comment (liamjohnston) [inline ietf/static_src/css/iab.scss]: css vars would of course make this a bit tricker - you'd need a separate variable for blue with x opacity, as an rgba colour i guess.

File: ietf/static_src/css/iab.scss [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+            rgba(62, 150, 200, 0) 100%
+        );
+        // copied
+        content: '';
+        bottom: 0;
+        left: 0;
+        pointer-events: none;
+        position: absolute;
+        right: 0;
+        top: 0;
```
Comment (liamjohnston) [inline ietf/static_src/css/iab.scss]: These bottom/left/right/top of 0 can just be replaced with `inset: 0`

File: ietf/static_src/css/iab.scss [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+        &:hover {
+            color: $blue-2 !important;
+        }
+    }
+}
+
+.card-text-iab {
+    font-size: 16px;
+    line-height: 21px;
+    /* or 131% */
```
Comment (liamjohnston) [inline ietf/static_src/css/iab.scss]: Did this get copied from figs? This should be set as `1.31` (unitless). It's much more forgiving if you change the size later but not the line-height.

File: ietf/static_src/css/iab.scss [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+    a {
+        color: $brand-blue !important;
+        &:hover {
+            color: $blue-2 !important;
+        }
+    }
+}
+
+.card-text-iab {
+    font-size: 16px;
```
Comment (liamjohnston) [inline ietf/static_src/css/iab.scss]: `font-size` should use rems, for better accessibility when it comes to honouring users browser preferences.

Does this project have a px to rem function? I can send you some code we usually use if not.

File: ietf/static_src/css/iab.scss [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+    letter-spacing: 0.4px;
+    font-weight: 300;
+}
+
+.date-text-iab {
+    color: $navy;
+    font-size: 14px;
+    line-height: 18px;
+    /* identical to box height, or 129% */
+    letter-spacing: 0.4px;
```
Comment (liamjohnston) [inline ietf/static_src/css/iab.scss]: See other comments about rems for font-size and unitless for line heights. Applies to all the CSS.

File: ietf/static_src/css/iab.scss [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+        .btn {
+            background-color: $blue-2;
+            // border-radius: 36px;
+        }
+    }
+
+    .dropdown-menu-iab {
+        margin-top: 36px !important;
+        border-radius: 4px;
+    }
```
Comment (liamjohnston) [inline ietf/static_src/css/iab.scss:156]: Does this selector need to be nested?

File: ietf/static_src/css/main.scss [hunk 11]
```diff
@@ -1,4 +1,22 @@
+@import 'bootstrap/scss/functions';
+@import 'bootstrap/scss/variables';
+
+$custom-spacers: (
```
Comment (liamjohnston) [inline ietf/static_src/css/main.scss]: Everything else in this entry point file is an `@import`. Maybe the spacer stuff should move to a separate file that you import here?

File: ietf/templates/includes/footer-iab.html [hunk 12]
```diff
@@ -0,0 +1,35 @@
+<div 
+    style="
+        background-image: url('static/img/background-dark.jpg');
+        height: 48px;
```
Comment (liamjohnston) [inline ietf/templates/includes/footer-iab.html]: The background-image as an inline style I get (easier than dealing with assets in CSS). But maybe the height could be moved to a CSS file?

File: ietf/templates/includes/footer-iab.html [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+            </li>
+            <li class="navbar-text-iab footer-link-iab p-2">
+                <a href="https://youtube.com/playlist?list=PLC86T-6ZTP5jeMlmiwMRcJ68N5IlWpxuQ" rel="me">
+                    <img src="/static/img/youtube-icon.svg"></img>
+                </a>
+            </li>
+            <li class="navbar-text-iab footer-link-iab p-2">
+                <a href="https://mastodon.online/@intarchboard" rel="me">
+                    <img src="/static/img/mastodon-icon.svg"></img>
+                </a>
```
Comment (liamjohnston) [inline ietf/templates/includes/footer-iab.html]: TIL `rel="me"`.

Comment (liamjohnston) [reply ietf/templates/includes/footer-iab.html]: These links have no accessible text. And the images have no alt attribute. 

You should probably:
- add `aria=label="Twitter"` (etc) to each link, and
- add empty alt attribute (`alt=""`) to each img

Comment (liamjohnston) [reply ietf/templates/includes/footer-iab.html]: Also, image tags are usually self-closing (``)though I don't know how much that matters 🤷

File: ietf/templates/includes/header_iab.html [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+            </div>
+
+            <!-- Mobile search -->
+            <button
+                class="btn btn-link nav-link text-light ms-auto d-lg-none"
+                type="button"
+                data-bs-toggle="collapse"
+                data-bs-target="#navbarSearch"
+                aria-controls="navbarSearch"
+                aria-expanded="false"
```
Comment (liamjohnston) [inline ietf/templates/includes/header_iab.html:32]: I'm assuming the aria-expanded value gets updated on click? Is that handled by bootstrap JS?

File: ietf/templates/includes/header_iab.html [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+                <ul
+                    class="navbar-nav col-12 col-xl-auto justify-content-xl-center flex-grow"
+                >
+                {% for item in MENU %}
+                <li class="nav-item nav-item-iab btn-group me-2 {% if item.pk == self.pk %} active{% endif %}">
+                    <a
+                        href="{{ item.url }}"
+                        class="nav-link btn btn-brand-blue text-white navbar-text-iab {% if item.subitems %} pe-1{% endif %}"
+                        
+
```
Comment (liamjohnston) [inline ietf/templates/includes/header_iab.html]: Need all this whitespace in the tag?

File: ietf/static_src/css/iab.scss [hunk 16]
```diff
@@ -0,0 +1,200 @@
...
+$navy: #002d3c; // (0, 45, 60)
+$yellow: #ffd13e; // (255, 209, 62)
+
+$grey-1: #f7fafc; // (247, 250, 252)
+$grey-2: #f0f6fa; // (240, 246, 250)
+$grey-3: #e6edf2; // (230, 237, 242)
+$grey-4: #d9e0e5; // (217, 224, 229)
+$grey-5: #cbd2d6; // (230, 210, 214)
+
+.h1-iab {
```
Comment (liamjohnston) [inline ietf/static_src/css/iab.scss]: Is the `*-iab` classname suffix new, or an existing pattern?

I see you have an element with `btn-secondary-iab` but that element doesn't also have `btn-secondary`. 

But there's also an element with `container` (the bootstrap class) AND `container-iab`.

I'm trying to understand how this naming convention works. How (or if) BS overrides are distinct from all-new code.

**Review by liamjohnston — COMMENTED** [review]
Mostly looks great :) 🏆 

There are some big changes here, so I'd recommend running the axe browser plugin over this at different resolutions and in different states (menu open/closet etc), to make sure nothing new's creeped in.

I have some general questions/comments. But the only think I'd like to see fixed is the font size in rems (HMU if you want a sass function, I have it handy).

File: ietf/static_src/css/iab.scss (hunk 16 above)
Comment (CliffRobinson) [reply ietf/static_src/css/iab.scss]: I've gone through and made that consistent. Now classes that start with `iab-*` are standalones that apply iab specific styling, and `*--iab` are modifier classes intended to be applied elements that also have their non iab equivalent parent class applied.

File: ietf/templates/includes/header_iab.html (hunk 15 above)
Comment (CliffRobinson) [reply ietf/templates/includes/header_iab.html]: removed

File: ietf/templates/includes/footer-iab.html (hunk 13 above)
Comment (CliffRobinson) [reply ietf/templates/includes/footer-iab.html]: all those changes are made

File: ietf/templates/includes/footer-iab.html (hunk 12 above)
Comment (CliffRobinson) [reply ietf/templates/includes/footer-iab.html]: changed

File: ietf/static_src/css/main.scss (hunk 11 above)
Comment (CliffRobinson) [reply ietf/static_src/css/main.scss]: moved

File: ietf/static_src/css/iab.scss (hunk 10 above)
Comment (CliffRobinson) [reply ietf/static_src/css/iab.scss:156]: as per our discussion, trying to keep it specific

File: ietf/static_src/css/iab.scss (hunk 9 above)
Comment (CliffRobinson) [reply ietf/static_src/css/iab.scss]: changed

File: ietf/static_src/css/iab.scss (hunk 8 above)
Comment (CliffRobinson) [reply ietf/static_src/css/iab.scss]: changed using the function you provided :-D

File: ietf/static_src/css/iab.scss (hunk 7 above)
Comment (CliffRobinson) [reply ietf/static_src/css/iab.scss]: changed

File: ietf/static_src/css/iab.scss (hunk 6 above)
Comment (CliffRobinson) [reply ietf/static_src/css/iab.scss]: changed

File: ietf/static_src/css/iab.scss (hunk 5 above)
Comment (CliffRobinson) [reply ietf/static_src/css/iab.scss]: left as is per our discussion

File: ietf/static_src/css/iab.scss (hunk 4 above)
Comment (CliffRobinson) [reply ietf/static_src/css/iab.scss]: changed

File: ietf/static/img/linkedin-icon.svg (hunk 3 above)
Comment (CliffRobinson) [reply ietf/static/img/linkedin-icon.svg:1]: leaving it as is per our discussion

File: ietf/static/img/iab-logo.svg (hunk 2 above)
Comment (CliffRobinson) [reply ietf/static/img/iab-logo.svg]: changed

File: ietf/home/templates/includes/post_iab.html (hunk 1 above)
Comment (CliffRobinson) [reply ietf/home/templates/includes/post_iab.html:15]: cool, leaving it as is to stay more consistent with IETF

**Review by liamjohnston — APPROVED** [review]
Excellent, well done :)

**Pull request merged**